src/plugins/analysis/ipc/code/ipc_analyzer.py

```diff
@@ -3,44 +3,69 @@
 import json
 import tempfile
 from pathlib import Path
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, List, Union
 
 from docker.types import Mount
+from pydantic import BaseModel, Field
+from semver import Version
 
-from analysis.PluginBase import AnalysisBasePlugin
+from analysis.plugin import AnalysisPluginV0
+from analysis.plugin.compat import AnalysisBasePluginAdapterMixin
 from helperFunctions.docker import run_docker_container
 
 if TYPE_CHECKING:
-    from objects.file import FileObject
+    from io import FileIO
 
 DOCKER_IMAGE = 'ipc'
 
 
-class AnalysisPlugin(AnalysisBasePlugin):
-    """
-    Inter-Process Communication Analysis
-    """
+class IpcTarget(BaseModel):
+    name: str = Field(description='target of the IPC (usually a file path or address)')
+    ipc_calls: List[IpcCall]
 
-    NAME = 'ipc_analyzer'
-    DESCRIPTION = 'Inter-Process Communication Analysis'
-    VERSION = '0.1.1'
-    FILE = __file__
 
-    MIME_WHITELIST = [  # noqa: RUF012
-        'application/x-executable',
-        'application/x-object',
-        'application/x-sharedlib',
-    ]
-    DEPENDENCIES = ['file_type']  # noqa: RUF012
-    TIMEOUT = 600  # 10 minutes
+class IpcCall(BaseModel):
+    type: str = Field(description='a type of call (usually a POSIX system call)')
+    arguments: List[Union[str, int, List[Union[str, int]]]] = Field(description='arguments of the IPC call')
```

Comment:
I do not think this field is useful in its current form. The typehint is too complex.
Wouldn't a `list[Union[str, int]]` suffice? Why have a special case for single arguments?

Reply:
> Wouldn't a list[Union[str, int]] suffice?

No because it can be a str, an int, a list of strings and a list of ints (so the type hint wasn't even technically correct; the correct one would be `List[Union[str, int, List[str], List[int]]]`)

Maybe we can just use a `: list`? :see_no_evil: